services/QuillLMS/app/queries/prompt_feedback_history.rb

```diff
@@ -18,36 +25,126 @@ def self.promptwise_sessions(activity_id)
 
     def self.promptwise_postprocessing(grouped_feedback_histories)
         prompt_hash = {}
-
+        
         prompts = Comprehension::Prompt.where(id: grouped_feedback_histories.map(&:prompt_id))
    
         grouped_feedback_histories.each do |prompt_session|
-            prompt_id = prompt_session.prompt_id
+            prompt_id = prompt_session.prompt_id.to_s.to_sym
 
             prompt_hash[prompt_id] ||= {
-                optimal_final_attempts: 0, 
-                session_count: 0, 
-                total_responses: 0,
-                final_attempt_pct_optimal: 0,
-                final_attempt_pct_not_optimal: 0,
-                display_name: ''
+
+                total_responses: 0.0,
+                session_count: 0.0, 
+                optimal_attempt_array: [],
+                display_name: '',
+
+                pct_final_attempt_optimal: 0.0,
+                pct_final_attempt_not_optimal: 0.0,
+                num_final_attempt_optimal: 0.0,
+                num_final_attempt_not_optimal: 0.0,
+
+                avg_attempts_to_optimal: 0.0,
+
+                num_consecutive_repeated_attempts_for_same_rule: 0.0,
+                num_non_consecutive_repeated_attempts_for_same_rule: 0.0,
+                pct_consecutive_repeated_attempts_for_same_rule: 0.0,
+                pct_non_consecutive_repeated_attempts_for_same_rule: 0.0,
+
```

Comment:
These names don't seem to me like they accurately reflect the measure - they're a count/percentage of sessions, not attempts. Maybe something more like `num_sessions_with_consecutive_repeated_rule`, etc?

Reply:
👍